src/table/utils/get-view-state.ts

```diff
@@ -0,0 +1,41 @@
+import { findPaginationVisibleRows, findVirtualizedVisibleRows } from './find-visible-rows';
+import getVisibleHeight from './get-visible-height';
+import { getTotalPosition } from '../../handle-data';
+import renderAsPagination from '../../render-as-pagination';
+import { TableLayout, ViewService } from '../../types';
+
+const getViewState = (layout: TableLayout, viewService: ViewService, rootElement: HTMLElement) => {
```

Comment:
Any changes to this function? hard to know when it has been moved

Reply:
There is a small change to this function. I move it to a separated file to make it easier to write unit tests and mock it in other unit files.